.github/workflows/trivy.yml

```diff
@@ -2,39 +2,45 @@ name: Trivy vulnerability scanner
 on:
   push:
     branches:
-      - master
+      - main
   pull_request:
 jobs:
-  build:
-    name: Build
+  build-and-scan:
+    name: Scan ${{ matrix.image }}
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - target: docker-build-workspace
+            image: test/workspace:latest
+            registry: test
+          - target: docker-build-ragengine
+            image: test/ragengine:latest
+            registry: test
+          - target: docker-build-ragservice
+            image: test/ragservice:latest
+            registry: test
+
     steps:
-      - name: Set up Go 1.x
-        uses: actions/setup-go@v5
-        with:
-          go-version: 1.24.2
-        id: go
 
       - name: Checkout code
-        uses: actions/checkout@v4
-      
-      - name: Build workspace image from Dockerfile
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Build image from Dockerfile
         run: |
-          export REGISTRY=test
+          export REGISTRY=${{ matrix.registry }}
           export VERSION=latest
-          export DOCKER_CLI_EXPERIMENTAL=enabled
           export OUTPUT_TYPE="type=docker"
           export ARCH="amd64"
-          make docker-build-workspace
-      
+          make ${{ matrix.target }} BUILD_FLAGS="--target dependencies"
```

Comment:
The image tags from `make ${{ matrix.target }}` might not be consistent with the ones you defined in the matrix